cmd/subnetcmd/upgradecmd/generate_upgrade.go

```diff
@@ -0,0 +1,379 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package upgradecmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/prompts"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/ava-labs/avalanche-cli/pkg/vm"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/storage"
+	"github.com/ava-labs/subnet-evm/commontype"
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/spf13/cobra"
+)
+
+type PrecompilePrompt interface {
+	PromptParams() error
+	ToMap() map[string]interface{}
+}
+
+type contractAllowList struct {
+	enabledAddresses []common.Address
+	adminAddresses   []common.Address
+}
+
+type feeManager struct {
+	adminAddresses   []common.Address
+	enabledAddresses []common.Address
+	initialFeeConfig commontype.FeeConfig
+}
+
+type nativeMint struct {
+	adminAddresses   []common.Address
+	enabledAddresses []common.Address
+	initialMint      map[string]string
+}
+
+type txAllowList struct {
+	enabledAddresses []common.Address
+	adminAddresses   []common.Address
+}
+
+const (
+	blockTimestampKey   = "blockTimestamp"
+	feeConfigKey        = "initialFeeConfig"
+	initialMintKey      = "initialMint"
+	adminAddressesKey   = "adminAddresses"
+	enabledAddressesKey = "enabledAddresses"
+
+	enabledLabel = "enabled"
+	adminLabel   = "admin"
+)
+
+// avalanche subnet upgrade generate
+func newUpgradeGenerateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "generate [subnetName]",
+		Short: "Generate the configuration file to upgrade subnet nodes",
+		Long:  `Upgrades to subnet nodes can be executed by providing a upgrade.json file to the nodes. This command starts a wizard guiding the user generating the required file.`,
```

Comment:
OK I will. TBH though while this may improve readability of the code, it looks odd to the user IMO, because we have to break up where you usually wouldn't.

Reply:
If you don't break it up, I don't think it wraps at word boundaries. It will just cut off at whatever character is at the end of the screen.

Reply:
OK. In any case it is now broken up.